A36-xds-for-servers.md

```diff
@@ -197,8 +198,7 @@ on `use_original_dst`. It also applies to `server_names`, `transport_protocol`,
 
 The overall "wrapping" server API has many language-specific ramifications. We
 show each individual language's approach. For C++ and wrapped languages we just
-show a rough sketch of how they would be done and expect a followup gRFC for
-more implementation details.
+show a rough sketch of how they would be done.
```

Comment:
@yashykt, do we have enough details for the C++ API at this point that we can include them here?

Reply:
Yeah, I think the C++ API is decided. We can add that here or create a separate gRFC. I'm fine with either of the two.

Reply:
Let's include it here.

Reply:
This PR is pretty close, so let's do the C++ as a followup PR. It is just as easy, if not easier (you don't have to push to my branch), that way.